### PR TITLE
Pull images with podman

### PR DESCRIPTION
**Pull request by henryprecheur**
Provide basic salinity check for images. The work to fix credentials is forthcoming.

### CODE REVIEW

File: pkg/runtime/podman_test.go [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+	return exists
+}
+
+// This test takes a long time to execute. Around 14 second on my computer.
+func TestPullImageNoCredentials(t *testing.T) {
+	if !*testPodman {
+		return
+	}
+
+	// XXX find a smaller image
```
Comment (hidalgopl) [inline pkg/runtime/podman_test.go:28]: alpine is around 6 megs:
```
✗ docker images | grep alpine
alpine                                                                     latest                                     d6e46aa2470d   2 months ago    5.57MB
alpine                                                                     3.9                                        78a2ce922f86   8 months ago    5.55MB
```

Comment (henryprecheur) [reply pkg/runtime/podman_test.go]: docker.io/library/hello-world is only 6kb ;-)

**Comment by henryprecheur** [comment]
Fixes #117

**Comment by henryprecheur** [comment]
I updated libpod to 2.2.1 to make it work with podman 2.2+

File: README.md [hunk 2]
```diff
@@ -12,3 +12,9 @@ Itzo is the agent that runs on [Kip](https://github.com/elotl/kip) cells (cloud
 ## Creating Images
 
 Scripts and a packer config building cell images on AWS and GCP can be found [here](https://github.com/elotl/kip-cell-image).
+
+## Podman < 2.2 required
+
+There’s an [issue][0] with libpod and Podman 2.2+. Podman 2.0 works fine.
+
+[0]: https://github.com/elotl/kip-cell-image/issues/6
```
Comment (henryprecheur) [inline README.md:20]: I’m going to remove this. It’s outdated. Ignore.

File: go.mod [hunk 3]
```diff
@@ -7,8 +7,7 @@ require (
 	github.com/aws/aws-sdk-go v1.28.2
 	github.com/containerd/cgroups v0.0.0-20190919134610-bf292b21730f
 	github.com/containernetworking/cni v0.8.0 // indirect
-	github.com/containers/buildah v1.17.0 // indirect
-	github.com/containers/libpod/v2 v2.0.4
+	github.com/containers/podman/v2 v2.2.1
```
Comment (henryprecheur) [inline go.mod:10]: Use most recent 'libpod' which is now 'podman' there’s a redirection, and the lastest version is declared as 'github.com/containers/podman' therefor the need for name change.

File: pkg/runtime/podman.go [hunk 4]
```diff
@@ -144,11 +147,11 @@ func (pcs *PodmanContainerService) CreateContainer(unit api.Unit, spec *api.PodS
 	container := convert.UnitToContainer(unit, nil)
 	// TODO: ignore this failure for now, it seems that there's response serialization bug on podman site
```
Comment (hidalgopl) [inline pkg/runtime/podman.go:148]: This is no longer accurate thanks to your changes;)

File: pkg/runtime/podman_test.go [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+
+func image_exists(conn context.Context, imageName string) bool {
+	exists, err := images.Exists(conn, imageName)
+	if err != nil {
+		panic(err)
+	}
+	return exists
+}
+
+// This test takes a long time to execute. Around 14 second on my computer.
```
Comment (hidalgopl) [inline pkg/runtime/podman_test.go:22]: outdated, right?

**Review by hidalgopl — APPROVED** [review]
Looks good, thanks! :+1: 
checked it also in the cluster and works fine:
```
I0115 09:53:46.806984    1591 pod_controller.go:118] New pod update
I0115 09:53:46.807643    1591 pod_controller.go:259] syncing pod units...
I0115 09:53:46.808095    1591 pod_controller.go:262] detected change: pod_created
I0115 09:53:46.808125    1591 pod_controller.go:449] status units are nil, trying to create pod from scratch
I0115 09:53:46.949108    1591 server.go:185] primary & secondary: 10.128.0.56 & 10.128.0.73
I0115 09:53:47.818525    1591 pod_controller.go:455] trying to create container: web
I0115 09:53:47.851430    1591 podman.go:127] trying to pull image: nginx for container: web
Completed short name "nginx" with unqualified-search registries (origin: /etc/containers/registries.conf)
Trying to pull docker.io/library/nginx:latest...
I0115 09:53:48.032961    1591 server.go:185] primary & secondary: 10.128.0.56 & 10.128.0.73
Getting image source signatures
Copying blob sha256:f72584a26f32a61cd0e3968e4204f0c9043abf72fda53d47f7423f5eaaaca614
Copying blob sha256:0732ab25fa22b144865915d3eacc0552e30495549b9f897be8331511c3131ca6
Copying blob sha256:d7f36f6fe38fea767000aa5d275a6f3176b1be06b2e2aad67a24e32c560535e5
Copying blob sha256:a076a628af6f7dcabc536bee373c0d9b48d9f0516788e64080c4e841746e6ce6
Copying blob sha256:7125e4df9063a34f9ce6445faee4da15a0974a60605de80ea40402cd2aca9943
Copying config sha256:f6d0b4767a6c466c178bf718f99bea0d3742b26679081e52dbf8e0c7c4c42d74
Writing manifest to image destination
Storing signatures
I0115 09:53:59.750191    1591 podman.go:198] podman started unit web as container web with nginx image
I0115 09:53:59.750305    1591 pod_controller.go:463] successfully created pod default_nginx
```

File: pkg/runtime/podman_test.go [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+	if image_exists(conn, imageName) {
+		_, err = images.Remove(conn, imageName, false)
+		if err != nil {
+			panic(err)
+		}
+		if image_exists(conn, imageName) {
+			t.Fatalf("Image %s hasn't been removed", imageName)
+		}
+	}
+	p.imgPuller.PullImage(tmpdir, "container", imageName, nil, false)
```
Comment (hidalgopl) [inline pkg/runtime/podman_test.go:50]: we should probably check if no error is returned
```
err = p.imgPuller.PullImage(tmpdir, "container", imageName, nil, false)
assert.NoError(t, err)
```

**Pull request merged**